utils/clarity_extract_handler.py

```
@@ -0,0 +1,541 @@
+        The path to the clarity extract file.
+    Outputs:
+    clarity_df : pd.DataFrame
+        DataFrame containing the contents of the clarity extract file.
+    Raises:
+        ValueError: If the clarity extract file is empty.
+        Exception: If there is an error reading the clarity extract file.
+    """
+    try:
+        clarity_df = pd.read_csv(clarity, delimiter=",")
+    except pd.errors.EmptyDataError:
+        raise ValueError(f"Clarity extract file is empty: {clarity}")
```

Ruff (0.14.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)